src/rdysetup.c

```diff
@@ -123,6 +124,30 @@ static PetscErrorCode CreateDM(RDy rdy) {
   // read the grid from a file
   PetscCall(DMPlexCreateFromFile(rdy->comm, rdy->config.grid.file, "grid", PETSC_TRUE, &rdy->dm));
 
+  // if we're using CEED, set Vec and Mat types based on the selected backend
+  if (rdy->ceed_resource[0]) {
+    VecType vec_type = NULL;
+    MatType mat_type = NULL;
+
+    CeedMemType mem_type_backend;
+    PetscCallCEED(CeedGetPreferredMemType(rdy->ceed, &mem_type_backend));
+    switch (mem_type_backend) {
+      case CEED_MEM_HOST:
+        vec_type = VECSTANDARD;
+        break;
+      case CEED_MEM_DEVICE: {
+        const char *resolved;
+        PetscCallCEED(CeedGetResource(rdy->ceed, &resolved));
+        if (strstr(resolved, "/gpu/cuda")) vec_type = VECCUDA;
+        else if (strstr(resolved, "/gpu/hip")) vec_type = VECKOKKOS;
+        else if (strstr(resolved, "/gpu/sycl")) vec_type = VECKOKKOS;
+        else vec_type = VECSTANDARD;
+      }
+    }
+    PetscCall(DMSetMatType(rdy->dm, mat_type));
```

Comment:
The `mat_type` should be also set like https://github.com/CEED/libCEED/blob/main/examples/fluids/navierstokes.c#L150C3-L152

Reply:
Thanks, fixed!